### PR TITLE
Dead cyborgs now have a GPS signal

### PR DESCRIPTION
**Pull request by zxaber**
## About The Pull Request
Gives cyborgs killed (but still intact) a GPS signal, so that they can be found if someone bothers to look for them. The signal shows up as "Cyborg Distress Signal", and goes away if they get repaired and rebooted or dismantled.

## Why It's Good For The Game
Helps with recovering borgs lost in the far stretches of Lavaland or abandoned in maintenance somewhere. It's like suit sensors for for borgs.

## Changelog
:cl:
add: Dead borgs now have a GPS signal.
/:cl:

### CODE REVIEW

**Comment by Dawson1917** [comment]
Needing a toolbelt to deconstruct validhunting borgs whenever you kill them to avoid getting snitched on is no good

**Comment by itseasytosee** [comment]
I think this would be great if there was an easy way to disable it IE wire cutters on a dead cyborg. Its should be just as easy as removing some clothing

**Comment by zxaber** [comment]
I suppose I could make it require that the cell is in place. Would require a crowbar.

File: code/game/objects/items/robot/robot_upgrades.dm [hunk 1]
```diff
@@ -62,6 +62,9 @@
 	if(R.mind)
 		R.mind.grab_ghost()
 		playsound(loc, 'sound/voice/liveagain.ogg', 75, TRUE)
+		var/datum/component/gps/G = R.GetComponent(/datum/component/gps)
+		if(G)
+			G.RemoveComponent()
```
Comment (ninjanomnom) [inline code/game/objects/items/robot/robot_upgrades.dm]: You should be removing the gps regardless of if it has a mind

**Review by ninjanomnom — CHANGES_REQUESTED** [review]

**Comment by Shadowflame909** [comment]
I would agree with this, but without access theres no way to stop it. compared to how easy it is to remove suit sensors/ strip a suit.

**Comment by tiramisuapimancer** [comment]
> 
> 
> I would agree with this, but without access theres no way to stop it. compared to how easy it is to remove suit sensors/ strip a suit.

Don't dead borgs automatically unlock their cover?

**Comment by ATH1909** [comment]
IMO, a better solution would be to make dead borgs show up in the robotics console (along with their location (and the locations of living borgs, if those aren't already in there)).

**Comment by HomocidlePancakes** [comment]
You do realize you can just look through cameras to see where the borgs are, regardless of bring alive?

**Comment by kriskog** [comment]
Good, another reason to just blow them.

**Comment by AdamElTablawy** [comment]
krisog at it again with the bad takes

pancakes is correct tho, this is needless. unless the camera wire is snipped then the borg can be seen through cameras.

**Comment by IndieanaJones** [comment]
Honestly, what would probably be better is if dead cyborgs had a flashing light on them of sorts when they die.  The main issue I've noticed as a cyborg is most of the time people just ignore you when you're dead.

**Comment by HomocidlePancakes** [comment]
>Honestly, what would probably be better is if dead cyborgs had a flashing light on them of sorts when they die. The main issue I've noticed as a cyborg is most of the time people just ignore you when you're dead.

Same with normal humans. Player issue, not code.

**Comment by optimumtact** [comment]
you can use their camera yeah.

**Pull request closed without merging**